packages/styled-components/src/utils/flatten.js

```diff
@@ -64,20 +62,9 @@ export default function flatten(
 
   /* Either execute or defer the function */
   if (isFunction(chunk)) {
-    if (isStatelessFunction(chunk) && executionContext) {
-      const result = chunk(executionContext);
-
-      if (process.env.NODE_ENV !== 'production' && isElement(result)) {
-        // eslint-disable-next-line no-console
-        console.warn(
-          `${getComponentName(
-            chunk
-          )} is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.`
```

Comment:
I'm not sure I agree with fully removing this check. We have a util for detecting styled components that could be used here to get a similar result.